tests/asp/test_communes.py

```diff
@@ -75,3 +76,25 @@ def test_by_insee_code_and_period(self):
 
         for period in [new_commune.start_date, datetime.date(2022, 11, 28), timezone.localdate()]:
             assert Commune.objects.by_insee_code_and_period(99999, period) == new_commune
+
+
+@pytest.mark.parametrize(
+    "insee_code,expected",
+    [
+        ("13150", "013"),  # Tarascon dans les Bouches-du-Rhône
+        ("32389", "032"),  # Saint-Martin dans le Gers
+        ("97416", "974"),  # Saint-Pierre à La Réunion
+        ("97502", "975"),  # Saint-Pierre à Saint-Pierre-et-Miquelon
+        ("97127", "971"),  # Saint-Martin en Guadeloupe
+    ],
+)
+def test_department_code(insee_code, expected):
+    commune = Commune(code=insee_code)
+    assert commune.department_code == expected
+
+
+def test_manager():
+    Commune.objects.all().delete()
+    Commune.objects.create(code="97411", start_date=datetime.date(1940, 1, 1), name="Saint-Denis", ignore=True)
+    assert Commune.objects.count() == 0
+    assert Commune.unfiltered_objects.count() == 1
```

Comment:
Le `.delete()` me fait un peu peur :sweat_smile:, donc proposition alternative :
```suggestion
def test_manager(faker):
    Commune.objects.create(code="99999", start_date=datetime.date.min, name=fake.word(), ignore=True)
    assert Commune.unfiltered_objects.count() == Commune.objects.count() + 1
```

Mais le problème que je vois c'est que si les fixtures ajoutent une commune ignorée alors le test va échouer donc peut-être plutôt :
```suggestion
def test_manager(faker):
    Commune.objects.create(code="99999", start_date=datetime.date.min, name=fake.word(), ignore=True)
    assert Commune.objects.filter(code="99999").count() == 0
    assert Commune.unfiltered_objects.filter(code="99999").count() == 1
```

Ce qui permettrais donc de faire ceinture-bretelle en modifiant `test_asp_INSEE_communes_factory.json` pour lui ajouter une commune ignorée (et la boucle est bouclée :D) :
```json
   {
      "model": "asp.Commune",
      "pk": 76198,
      "fields": {
         "code": "97811",
         "name": "SAINT-DENIS",
         "start_date": "1900-01-01",
         "end_date": null,
         "ignore": true,
      }
   },
```

Reply:
Merci pour le code !